package/src/prmon.cpp

```diff
@@ -266,6 +284,12 @@ int main(int argc, char* argv[]) {
         << (default_store_hw_info ? "true" : "false") << ")\n"
         << "[--netdev, -n dev]        Network device to monitor (can be given\n"
         << "                          multiple times; default ALL devices)\n"
+        << "[--monitor, -m mon]       Enable or disable monitors:\n"
+        << "                           ~NAME disables that monitor\n"
+        << "                            NAME enables that monitor\n"
+        << "                          comma separation supported or specifing\n"
+        << "                          multiple times\n"
+        << "                          Special name '[~]all' sets default state\n"
```

Comment:
I love the idea but I'm a bit conflicted w/ the set of rules we have. 

For example, if as a user I'd like to only monitor `cpumon`, my initial reaction would be to execute `prmon -m cpumon -- [command]`. However, doing so I end up getting everything (w/o any message) because I should've used a rather awkward `prmon -m ~all,cpumon -- [command]`.

Can we do something like this instead:

- If no `-m` is provided collect everything (default as is now)
- Don't allow mixing `~` and non-`~`
    - If the default is `all` then one can turn a subset off things w/ a simple `-m ~a,~b`
    - If the user provides `-m a,b` then only enable those monitors

In all cases, we better print exactly what's going to be monitored. Currently we only print a message saying `wallmon` cannot be turned-off if the user tries to do so.

Then there are edge cases, which can be blamed on the user, but `-m ~all,all` ends up collecting only `wallmon` but `-m all,~all` ends up collecting all (i.e. the options are not commutative). Perhaps one can print a warning in such cases. If a simple `-m ~all` is provided (one can argue that's silly but) we can simply bail out printing a message that the user asked for monitoring nothing.

Also another issue is that, w/ this approach we're altering the orders of the columns in the text file. Now this is not really a problem on our side per se (for example the plotting script uses column names). However, ADC is making plots using these files and if they have hardcoded column numbers we should make sure they're aware. Technically it's someone else's problem but would be nice to give a heads up.

Reply:
I like your proposal for the syntax - I was probably thinking too generally and then it does become a bit confusing (especially with `all` being the implicit default).

The issue with `~all,all != all,~all` is just impossible to solve satisfactorily, but is avoided if we disallow mixing on/off arguments.

Re. the columns, we should check that they use column names as we can't really guaranttee any particular ordering.